build.gradle.kts

```diff
@@ -0,0 +1,280 @@
+import ktx.*
+import org.gradle.api.tasks.testing.logging.TestExceptionFormat.FULL
+import org.gradle.api.tasks.testing.logging.TestLogEvent.*
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+buildscript {
+  repositories {
+    jcenter()
+    mavenCentral()
+  }
+
+  val dokkaVersion: String by project
+  val kotlinVersion: String by project
+  val junitPlatformVersion: String by project
+  val configurationsPluginVersion: String by project
+
+  dependencies {
+    classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion")
+    classpath("com.netflix.nebula:gradle-extra-configurations-plugin:$configurationsPluginVersion")
+    classpath("org.jetbrains.dokka:dokka-gradle-plugin:$dokkaVersion")
+    classpath("org.junit.platform:junit-platform-gradle-plugin:$junitPlatformVersion")
+  }
+}
+
+val libGroup: String by project
+val ossrhUsername: String by project
+val ossrhPassword: String by project
+
+plugins {
+  java
+  distribution
+  id("io.codearte.nexus-staging") version "0.22.0"
+}
+
+repositories {
+  jcenter()
+}
+
+val libVersion = file("version.txt").readText().trim()
+
+allprojects {
+  val linter = configurations.create("linter")
+
+  dependencies {
+    linter("com.pinterest:ktlint:$ktlintVersion")
+  }
+}
+
+subprojects {
+  apply(plugin = "maven-publish")
+  apply(plugin = "java")
+  apply(plugin = "kotlin")
+  apply(plugin = "signing")
+  apply(plugin = "org.jetbrains.dokka")
+  apply(plugin = "jacoco")
+  apply(plugin = "nebula.provided-base")
+
+  val isReleaseVersion = !libVersion.endsWith("SNAPSHOT")
+
+  repositories {
+    mavenLocal()
+    jcenter()
+    mavenCentral()
+    maven("https://oss.sonatype.org/content/repositories/snapshots/")
+  }
+
+  group = libGroup
+  version = libVersion
+  val projectName: String by project
+  val projectDesc: String by project
+
+  java {
+    sourceCompatibility = JavaVersion.VERSION_1_6
+    targetCompatibility = JavaVersion.VERSION_1_6
+  }
+
+  tasks.withType<KotlinCompile> {
+    kotlinOptions {
+      jvmTarget = "1.6"
+      freeCompilerArgs += "-Xopt-in=kotlin.RequiresOptIn"
+    }
+  }
+
+  dependencies {
+    val kotlinVersion: String by project
+
+    "provided"("org.jetbrains.kotlin:kotlin-stdlib:$kotlinVersion")
+    "provided"("com.badlogicgames.gdx:gdx:$gdxVersion")
```

Comment:
Pretty sure `nebula.provided-base` plugin is no longer required and you should use `compileOnly` in newer Gradle versions. Could you test if a similar POM is generated with `compileOnly`?

Reply:
Ok, I will take a look at that!

Reply:
Had to make some extra changes and discovered that the pom was not being correctly created D:

But know everything seems to be working as expected